File: book-crud-frontend/src/service/BookService.js
import axios from "axios";
const BOOK_BASE_URL = 'http://localhost:8080/api/books';
class BookService {
    getAllBooks(){
        return axios.get(BOOK_BASE_URL);
    }
    createBook(book){
        return axios.post(BOOK_BASE_URL, book);
    }
    getBookbyId(bookId){
        return axios.get(BOOK_BASE_URL + "/" + bookId);
    }
    updateBook(bookId, book){
        return axios.put(BOOK_BASE_URL + "/" + bookId, book);
    }
    deleteBook(bookId){
        return axios.delete(BOOK_BASE_URL + "/" + bookId);
    }
}
export default new BookService();

File: book-crud-backend/src/main/java/com/backend/service/BookService.java
package com.backend.service;

import com.backend.entity.Book;

import java.util.List;

public interface BookService {
    Book addBook(Book book);
    List<Book> getAllBooks();
    Book getBookById(long id);
    Book updateBook(long id, Book bookDetails);
    Book deleteBook(long id);
}
